pkg/controller/studyjob/const.go

```diff
@@ -12,12 +12,14 @@ limitations under the License.
 package studyjob
 
 const (
-	DefaultJobWorker      = "Job"
-	TFJobWorker           = "TFJob"
-	PyTorchJobWorker      = "PyTorchJob"
-	jobTypeNAS            = "NAS"
-	jobTypeHP             = "HP"
-	suggestionAlgorithmRL = "nasrl"
+	DefaultJobWorker = "Job"
+	TFJobWorker      = "TFJob"
+	PyTorchJobWorker = "PyTorchJob"
+	jobTypeNAS       = "NAS"
+	jobTypeHP        = "HP"
+	suggestionNASRL  = "nasrl"
 )
 
 var ValidWorkerKindList = [...]string{DefaultJobWorker, TFJobWorker, PyTorchJobWorker}
+
+var suggestionNASList = []string{suggestionNASRL}
```

Comment:
Our goal is to keep controller independent of worker type and suggestion algos.  Currently there are no worker specific code in the code(except for watch. Hope, it can be fixed soon and ValidWorkerKindList won't not needed any more).  Similarly i feel that there should not be any suggestion algorithm specific changes in the controller code.

Reply:
In that case, how we can make a validation for the NAS job ?